### ISSUE TITLE
Processing of whitespaces before brackets by NodeSet

### ISSUE THREAD

**Issue by bougetq** [cea-hpc/clustershell#268]
On the current version of the master branch whitespaces are removed before an opening bracket:
```python
>>> [node for node in NodeSet('node [0]')]
['node0']
```

But not elsewhere:
```python
>>> [node for node in NodeSet('node 0')]
['node 0']
```

On the version 1.6 of ClusterShell whitespaces were not removed before brackets.
This issue appears from this commit: e49ad6344ca4868e4557003a6b7e7a74d30fd248

**Comment by thiell** [comment]
Thanks, good catch. Feel free to assign the ticket to yourself and submit a patch if you have time. If not, I will have a look at it as soon as I'm done with some other tickets.

**Comment by bougetq** [comment]
I would gladly assign the ticket to myself but I cannot do it myself.

**Comment by thiell** [comment]
No worries.

Open question: what behavior should we expect from NodeSet in that case? Should spaces be just trimmed? Should we raise an error?

Currently we have the following related tests:
```python
    def testWhitespacePrefix(self):
        """test NodeSet parsing ignoring whitespace"""
        nodeset = NodeSet(" tigrou2 , tigrou7 , tigrou[5,9-11] ")
        self.assertEqual(str(nodeset), "tigrou[2,5,7,9-11]")
        nodeset = NodeSet("   tigrou2 ,    tigrou5,tigrou7 , tigrou[ 9   - 11 ]    ")
        self.assertEqual(str(nodeset), "tigrou[2,5,7,9-11]")
```

So to be coherent, I would say spaces should just be trimmed... I wonder it's it the best thing to do however. Let's think about it before I start working on this.

**Comment by bougetq** [comment]
naparuba/shinken actually has feature that needs whitespaces inside the node name to be preserved.
I have a patch to do this -- trim whitespaces before and after a node name while preserving the ones inside. I tried to push it on Gerrit to run the full test suite but it failed (the pushing part) so I think I might need to read some more documentation.

**Comment by thiell** [comment]
You should be able to push on it on the refs/for/master branch for review, assuming you got a gerrithub login.

gerrithub url = ssh://yourgerrithublogin@review.gerrithub.io:29418/cea-hpc/clustershell

I will be interested if that works for you so that I can document it somewhere.

**Comment by bougetq** [comment]
It worked, I had to install the hook to insert Change-Id but it was really easy since there was an error message explaining what to do.

**Comment by degremont** [comment]
IMO, current tests are OK and we should keep this behaviour.
The question is: "what should we do for the following patterns?"
* ```name with some space```
* ```space before range [1-10]```

In current master code. The first one is left untouched, the second got space trimmed before and after the range brackets. This sounds weird. Let's simply restore the 1.6 behaviour for now.

**Comment by thiell** [comment]
@degremont: ok for me to keep the 1.6 behavior
@bougetq: please assign this ticket to yourself :)

https://review.gerrithub.io/241255

**Comment by thiell** [comment]
@bougetq: oh and thanks for the feedback about gerrithub. Once we have the new docs ready, I will add a short section explaining this!

**Comment by bougetq** [comment]
I do not have the permissions to self-assign a ticket.

**Comment by thiell** [comment]
Closed by 71f4a6cf7e106f2d6d836dc5b681e4ad90d357b2 - thanks!

**Comment by bougetq** [comment]
You are welcome!
And thank you for your advice!
